### PR TITLE
FORUM-1377 Birthday shown in the Public View profile

### PR DESCRIPTION
**Pull request by seifeddineDaoud**

### CODE REVIEW

File: integ-forum/integ-forum-social/src/main/java/org/exoplatform/forum/ext/common/SocialContactProvider.java [hunk 1]
```diff
@@ -65,6 +65,9 @@ public CommonContact getCommonContact(String userId) {
         if (profile.contains(Profile.GENDER)) {
           contact.setGender(profile.getProperty(Profile.GENDER).toString());
         }
+        if (profile.contains(Profile.BIRTHDAY)) {
+            contact.setBirthday(profile.getProperty(Profile.BIRTHDAY).toString());
+        }
```
Comment (thomasdelhomenie) [inline integ-forum/integ-forum-social/src/main/java/org/exoplatform/forum/ext/common/SocialContactProvider.java]: Adding the birthday to the Social profile is a workaround IMO. We don't need the birthday on social side, we add it only to make it visible in forum.
The birthday is in the User Profile (yes there is a user profile and a social profile, for historical reasons. We aim to harmonize this, but that's another topic...), so instead we should update this contact provider SocialContactProvider to fetch the birthday where it already is:

```java
  UserProfile profile = orgService.getUserProfileHandler().findUserProfileByName(userId);
  if (profile.getUserInfoMap() != null) {
          contact.setBirthday(profile.getAttribute("user.bdate"));
  }
```

Comment (seifeddineDaoud) [reply integ-forum/integ-forum-social/src/main/java/org/exoplatform/forum/ext/common/SocialContactProvider.java]: Okay, I'll change it

File: integ-forum/integ-forum-social/src/main/java/org/exoplatform/forum/ext/common/SocialContactProvider.java [hunk 2]
```diff
@@ -51,6 +53,12 @@ public CommonContact getCommonContact(String userId) {
       if (userIdentity != null) {
         contact = new CommonContact();
         Profile profile = userIdentity.getProfile();
+        UserProfile userProfile = null;
+        try {
+          userProfile = CommonsUtils.getService(OrganizationService.class).getUserProfileHandler().findUserProfileByName(userId);
```
Comment (thomasdelhomenie) [inline integ-forum/integ-forum-social/src/main/java/org/exoplatform/forum/ext/common/SocialContactProvider.java]: OrganizationService should be injected instead:

```java
private OrganizationService organizationService;

public SocialContactProvider(OrganizationService organizationService) {
  this.organizationService = organizationService;
}
```

File: integ-forum/integ-forum-social/src/main/java/org/exoplatform/forum/ext/common/SocialContactProvider.java [hunk 3]
```diff
@@ -51,6 +53,12 @@ public CommonContact getCommonContact(String userId) {
...
+        } catch (Exception e) {
+          LOG.error("Could not retrieve forum user profile for " + userId + ": ", e);
```
Comment (thomasdelhomenie) [inline integ-forum/integ-forum-social/src/main/java/org/exoplatform/forum/ext/common/SocialContactProvider.java]: "Could not retrieve forum user profile for " -> "Could not retrieve user profile for "

File: integ-forum/integ-forum-social/src/main/java/org/exoplatform/forum/ext/common/SocialContactProvider.java [hunk 4]
```diff
@@ -65,6 +73,9 @@ public CommonContact getCommonContact(String userId) {
         if (profile.contains(Profile.GENDER)) {
           contact.setGender(profile.getProperty(Profile.GENDER).toString());
         }
+        if (userProfile.getUserInfoMap() != null) {
```
Comment (thomasdelhomenie) [inline integ-forum/integ-forum-social/src/main/java/org/exoplatform/forum/ext/common/SocialContactProvider.java]: Check for userProfile nullability also here.

**Pull request merged**